backend/iam/sso/oidc/views.py

```
@@ -0,0 +1,40 @@
+def callback(request, provider_id):
+    try:
+        response = OAuth2CallbackView.adapter_view(
+            OpenIDConnectOAuth2Adapter(request, provider_id)
+        )(request)
+        if response.status_code != 302:
+            return response
+        token = generate_token(request.user)
+        next = f"{response['Location'].rstrip('/')}/sso/authenticate/{token}"
+        return HttpResponseRedirect(next)
+    except SocialApp.DoesNotExist:
+        raise Http404
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)